### ISSUE TITLE
add .pls - error

### ISSUE THREAD

**Issue by mrbatcop** [y20k/transistor#178]
Hi!

I try to add this link
https://centova1.cheapshoutcast.com:2199/tunein/baroque.pls
But I see this message: 
Transistor tried hard but could not make sense of the link.
Hmm, what does it mean? I mean it is a .pls fault or it is a Transistor bug?
Update: Hmm .. It is interesting! I opened the link with the VLC then I copied the location of the stream to the Transistor and it works but very unstable - I mean it often stops playing! 

Thanks and Good Luck!

**Comment by y20k** [comment]
Thanks for finding/reporting this problem. I am AFK, but I will look into it. The PLS seems to be legit. ...

In the meantime try to add this URL manually. 

http://8.38.78.173:8045/stream

**Comment by y20k** [comment]
And about the instability... I don't know. That is probably out of my reach since I just hand the stream over to a audio playback library. But I will look what I can do.

**Comment by mrbatcop** [comment]
In the meantime try to add this URL manually.
=>
Thanks, but I already did this using VLC info, (See my previous post)

Good Luck!

**Comment by y20k** [comment]
Sorry that it took a bit longer. I was on vacation. 

The link https://centova1.cheapshoutcast.com:2199/tunein/baroque.pls is configured a bit unusual. Although the file is named `baroque.pls` the file you get is named `playlist.pls`. I am not sure, but I think **this is not** the problem, because the HTTP request does not reveal any redirects. 

But speaking of redirects ... Since I am onto this issue here, I adapted Transistor to handle redirects better. Just in case.

Even with the new redirect code I get an error. To be a bit more specific: `Trust anchor for certification path not found`. This looks like there is something wrong with the certificate of https://cheapshoutcast.com. I am not sure how to circumvent that. 

Interestingly **without the s** http://centova1.cheapshoutcast.com:2199/tunein/baroque.pls the station downloads just fine.

In short: As a result of your bug report I implemented a better handling of redirects. That does not solve your problem, but it make Transistor better. 😄 

And about the stuttering of the stream. I will try to increase the buffer and report back here.

**Comment by y20k** [comment]
> And about the stuttering of the stream. I will try to increase the buffer and report back here.

I just checked: I already did double the buffer a while ago. See [PlayerService.java#L703](https://github.com/y20k/transistor/blob/v3.0.4/app/src/main/java/org/y20k/transistor/PlayerService.java#L703). So now I am a bit reluctant to increase it even more. Because that would mean longer loading times.

I am not sure at all that the problem you experience is related to the buffer size, since I cannot reproduce it on my test phone. Here the stream just runs with the default double-bufferjust as fine as with a triple-buffer.

So. I am sorry @mrbatcop - but I guess I cannot help you here 😒.
